contract/r/gnoswap/gov/staker/v1/delegation.gno

```diff
@@ -86,46 +82,84 @@ func (r *DelegationResolver) Collect(amount, currentTime int64) error {
 
 // processCollection handles the actual collection logic
 func (r *DelegationResolver) processCollection(amount, currentTime int64) error {
-	remainingToCollect := amount
+	if amount < 0 {
```

Comment:
`amount` is assigned to `remaining` on line 89. Then `remaining` undergoes a `<= 0` check on line 105. Given the flow, it seems like a good idea to perform the `amount <= 0` check here.